packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -124,9 +124,24 @@ export function SelectPanel({
     }
   }, [placeholder, renderAnchor, selected])
 
+  const testIsItemSelected = React.useCallback((selectedItem: ItemInput, item: ItemInput) => {
+    const itemType = typeof selectedItem
+    if (itemType === 'object') {
+      if (selectedItem.hasOwnProperty('id')) {
+        return selectedItem.id === item.id
+      }
```

Comment:
I think we should also try testing for `key` since that's React's way of differentiating component changes

Reply:
I thought about that, but I actually think it's better to add the `key` when rendering over the list instead of trying to add it preemptively. Not super opinionated though if we want to use the key here!